source/gui/__init__.py

```diff
@@ -773,14 +794,18 @@ def __init__(self, parent):
 		# Translators: The label for actions list in the Exit dialog.
 		labelText=_("What would you like to &do?")
 		self.actions = [
-		# Translators: An option in the combo box to choose exit action.
-		_("Exit"),
-		# Translators: An option in the combo box to choose exit action.
-		_("Restart"),
-		# Translators: An option in the combo box to choose exit action.
-		_("Restart with add-ons disabled"),
-		# Translators: An option in the combo box to choose exit action.
-		_("Restart with debug logging enabled")]
+			# Translators: An option in the combo box to choose exit action.
+			_("Exit"),
+			# Translators: An option in the combo box to choose exit action.
+			_("Restart"),
+			# Translators: An option in the combo box to choose exit action.
+			_("Restart with add-ons disabled"),
+			# Translators: An option in the combo box to choose exit action.
+			_("Restart with debug logging enabled")
+		]
+		if updateCheck and updateCheck.isPendingUpdate():
+			# Translators: An option in the combo box to choose exit action.
+			self.actions.append(_("Run pending update"))

```

Comment:
Elsewhere we say "Install update" rather than "run update", I think "Install pending update" would be more consistent.